commit b22e8724ec88dcc60e594341a584bb0900c40699 (refs/changes/48/3048/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2018-03-07T00:07:21+00:00 (1 year, 7 months ago)
    
    OS-6497 Non-primary NICs of "joyent" branded zones don't work with just "addrconf"

diff --git a/src/vm/node_modules/VM.js b/src/vm/node_modules/VM.js
index 49210509..1cf3d63d 100644
--- a/src/vm/node_modules/VM.js
+++ b/src/vm/node_modules/VM.js
@@ -3247,6 +3247,10 @@ function writeZoneNetfiles(payload, log, callback)
                         static_ips);
                 }
 
+                if (addrconf_found) {
+                    replaceFile(zoneroot, '/etc/addrconf.' + n.interface, '');
+                }
+
                 if (n.hasOwnProperty('primary') && !primary_found) {
                     // Only allow one primary network
                     primary_found = true;
@@ -3257,10 +3261,6 @@ function writeZoneNetfiles(payload, log, callback)
                     if (dhcp_found) {
                         replaceFile(zoneroot, '/etc/dhcp.' + n.interface, '');
                     }
-                    if (addrconf_found) {
-                        replaceFile(zoneroot, '/etc/addrconf.' + n.interface,
-                            '');
-                    }
                 }
             });
         }
